src/lazy/text/buffer.rs

```diff
@@ -1889,6 +1963,15 @@ mod tests {
             r#"
             "this has an escaped quote \" right in the middle"
             "#,
+            r#" '''hi''' "#,
```

Comment:
Add case for empty string.
```suggestion
            r#" '''hi''' "#,
            r#" ''''''  '''''' "#,
```